cocoa/src/toga_cocoa/screen.py

```diff
@@ -0,0 +1,17 @@
+class Screen:
+    _screen_impl_cache = {}
+
+    def __new__(cls, native):
+        if native in cls._screen_impl_cache:
+            return cls._screen_impl_cache[native]
+        else:
+            instance = super().__new__(cls)
+            cls._screen_impl_cache[native] = instance
+            return instance
+
+    def __init__(self, native):
+        if not hasattr(self, "native"):
+            self.native = native
```

Comment:
I'm not wild about this as an implementation. Is there any reason not to fold this into the `__new__` implementation, and avoid having an `__init__` altogether?

Reply:
Now, there is only a `__new__` implementation.